tests/integration/test_interaction.py

```diff
@@ -674,6 +674,44 @@ def test_do_attempt_feedback_is_updated(self):
         self.assertEqual(self._get_feedback().text, expected_feedback)
 
 
+class AssessmentGradingTest(
+    DefaultAssessmentDataTestMixin, AssessmentTestMixin, InteractionTestBase, BaseIntegrationTest
+):
+    """
+    Testing Drag and Drop XBlock grading against default data in assessment mode.
+    If default data changes this will break.
+    """
+    def setUp(self):
+        mock = Mock()
+        context = patch.object(WorkbenchRuntime, 'publish', mock)
+        context.start()
+        self.addCleanup(context.stop)
+        self.publish = mock
+        super(AssessmentGradingTest, self).setUp()
+
+    def test_grade(self):
+        """
+        Test grading after submitting solution in assessment mode
+        """
+        self.place_item(0, TOP_ZONE_ID, Keys.RETURN) # Correctly placed item
+        self.place_item(1, BOTTOM_ZONE_ID, Keys.RETURN) # Incorrectly placed item
+        self.place_item(4, MIDDLE_ZONE_ID, Keys.RETURN) # Incorrectly placed decoy
+        self.click_submit()
+
+        # Find the grade event
+        published_grade = None
+        for call in self.publish.call_args_list:
```

Comment:
@arbrandes suggestion:

    grade_event = next(publish_call_args[0] for publish call in self.publish.call_args_list if publish_call_args[0][1] == 'grade')
    self.assertEqual(grade_event[2], expected_grade)

Not tested, but I hope you see what I mean here.

Alternatively, it might make sense to use `assertIn`:

    expected_event = (self.block, 'grade', expected_grade)
    self.assertIn(expected_event, self.publish_call_arg_list)